hikari/impl/rest.py

```diff
@@ -274,12 +291,18 @@ def acquire(self, token: str, token_type: str = _BEARER_TOKEN_PREFIX) -> rest_ap
         # Since we essentially mimic a fake App instance, we need to make a circular provider.
         # We can achieve this using a lambda. This allows the entity factory to build models that
         # are also REST-aware
-        entity_factory = entity_factory_impl.EntityFactoryImpl(_RESTProvider(lambda: rest_client))
+        cache = stateless_cache.StatelessCacheImpl()
+        provider = _RESTProvider(
+            lambda: entity_factory,
+            self._executor,
+            lambda: cache,
+            lambda: rest_client,
+        )
```

Comment:
Any reason why these have to be provided as lambdas?

Reply:
because circular dependency

Reply:
Ah, ok :+1: